openquake/hazardlib/gsim/zhao_2006.py

```diff
@@ -172,6 +197,13 @@ class ZhaoEtAl2006Asc(GMPE):
     #  DEFINED_FOR_REFERENCE_VELOCITY = 1100
     DEFINED_FOR_REFERENCE_VELOCITY = 800
 
+    def __init__(self, cb14_basin_term=False, m9_basin_term=False):
+        if cb14_basin_term or m9_basin_term:
```

Comment:
I think the issue is that we need the pre-basin term mean and then have to contextually swap the CB14 basin term with the M9 basin term depending on z2pt5 and period (based on the USGS java code for this model's "basin" variant - https://code.usgs.gov/ghsc/nshmp/nshmp-lib/-/blob/main/src/main/java/gov/usgs/earthquake/nshmp/gmm/ZhaoEtAl_2006.java#L270), which this approach allows.

The existing M9 and CB14 ModifiableGMPE classes can only obtain the mean + either basin term rather than contextually select the appropriate basin term within the compute method itself which is needed here.